Tasks/SonarQubePostTest/CodeAnalysisFilePathComputation.ps1

```
@@ -0,0 +1,231 @@
+$ProjectGuidAndFilePathMap = @{}
+$ComponentKeyAndPathMap = @{}
+$ComponentKeyAndRelativePathCache = @{}
+
+function ConstructComponentKeyAndPathMap($json)
+{
+    foreach ($component in $json.Components)
+    {
+        if (!$ComponentKeyAndPathMap.ContainsKey($component.key))
+        {
+            $ComponentKeyAndPathMap.Add($component.key, $component.path)
+        }
+    }
+}
+
+function IsComponentFormatValid($tokens)
+{
+    #expected format for component '[SonarQube project key]:[SonarQube project value]:[MSBuild project guid]:[file name relative to MSBuild project file path]'
+    $isFormatValid = $false
+
+    if (!$tokens)
+    {
+        Write-Verbose -Verbose "IsComponentFormatValid: tokens is invalid"
+        return $isFormatValid
+    }
+    if ($tokens.Count -ne 4) 
+    {
+        Write-Verbose -Verbose "IsComponentFormatValid: component is not in expected format, token count is not equal to 4"
+        return $isFormatValid
+    }
+
+    #third token must be a guid
+    $guidToken = $tokens[2]
+
+    $outGuid = New-Object -TypeName "System.Guid"
+    if (![System.Guid]::TryParse($guidToken, [ref]$outGuid))
+    {
+        Write-Verbose -Verbose "$guidToken is not a GUID"
```

same here, we want to fail silently in such scenarios and not fail the build..